src/storage/FileResourceStore.ts

```diff
@@ -0,0 +1,458 @@
+import arrayifyStream from 'arrayify-stream';
+import { ConflictHttpError } from '../util/errors/ConflictHttpError';
+import { contentType } from 'mime-types';
+import { DATA_TYPE_BINARY } from '../util/ContentTypes';
+import { ensureTrailingSlash } from '../util/Util';
+import { InternalServerError } from '../util/errors/InternalServerError';
+import { MethodNotAllowedHttpError } from '../util/errors/MethodNotAllowedHttpError';
+import { NotFoundHttpError } from '../util/errors/NotFoundHttpError';
+import { Readable } from 'stream';
+import { Representation } from '../ldp/representation/Representation';
+import { RepresentationMetadata } from '../ldp/representation/RepresentationMetadata';
+import { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { ResourceStore } from './ResourceStore';
+import streamifyArray from 'streamify-array';
+import { UnsupportedMediaTypeHttpError } from '../util/errors/UnsupportedMediaTypeHttpError';
+import { createReadStream, createWriteStream, promises as fsPromises, Stats } from 'fs';
+import { DataFactory, StreamWriter } from 'n3';
+import { InteractionHandler, RequestAction } from '../util/InteractionHandler';
+import { NamedNode, Quad } from 'rdf-js';
+
+/**
+ * Resource store storing its data in the file system backend.
+ * All requests will throw an {@link NotFoundHttpError} if unknown identifiers get passed.
+ */
+export class FileResourceStore implements ResourceStore {
+  private readonly base: string;
+  private readonly root: string;
+  private readonly interactionHandler: InteractionHandler;
+
+  private readonly rdf = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#';
+  private readonly ldp = 'http://www.w3.org/ns/ldp#';
+  private readonly terms = 'http://purl.org/dc/terms/';
+  private readonly xml = 'http://www.w3.org/2001/XMLSchema#';
+  private readonly stat = 'http://www.w3.org/ns/posix/stat#';
+
+  private readonly predicates = {
+    aType: DataFactory.namedNode(`${this.rdf}type`),
+    modified: DataFactory.namedNode(`${this.terms}modified`),
+    contains: DataFactory.namedNode(`${this.ldp}contains`),
+    mtime: DataFactory.namedNode(`${this.stat}mtime`),
+    size: DataFactory.namedNode(`${this.stat}size`),
+  };
+
+  private readonly objects = {
+    container: DataFactory.namedNode(`${this.ldp}Container`),
+    basicContainer: DataFactory.namedNode(`${this.ldp}BasicContainer`),
+    ldpResource: DataFactory.namedNode(`${this.ldp}Resource`),
+    dateTime: DataFactory.namedNode(`${this.xml}dateTime`),
+  };
+
+  /**
+   * @param base - Will be stripped of all incoming URIs and added to all outgoing ones to find the relative path.
+   * @param root - Root filepath in which the resources and containers will be saved as files and directories.
+   */
+  public constructor(base: string, root: string) {
+    this.base = base;
+    while (this.base.endsWith('/')) {
+      this.base = this.base.slice(0, -1);
+    }
```

Comment:
There can always be multiple though. Let's `this.base.replace(/\/+$/, '')` and test this case.
Or better, since this will be common, in a separate utility as a `trimTrailingSlashes` function like `ensureTrailingSlash`.